src/dsp_tools/commands/project/create/project_create.py

```diff
@@ -1120,17 +1089,55 @@ def create_project(
     # final steps
     if overall_success:
         msg = (
-            f"Successfully created project '{proj_shortname}' ({proj_shortcode}) with all its ontologies. "
+            f"Successfully created project '{project_definition.shortname}' "
+            f"({project_definition.shortcode}) with all its ontologies."
             f"There were no problems during the creation process."
         )
         print(f"========================================================\n{msg}")
         logger.info(msg)
     else:
         msg = (
-            f"The project '{proj_shortname}' ({proj_shortcode}) with its ontologies could be created, "
+            f"The project '{project_definition.shortname}' ({project_definition.shortcode}) "
+            f"with its ontologies could be created, "
             f"but during the creation process, some problems occurred. Please carefully check the console output."
         )
         print(f"========================================================\nWARNING: {msg}")
         logger.warning(msg)
 
     return overall_success
+
+
+def _prepare_project_information_onto_lists(
+    project_json: dict[str, Any],
+) -> tuple[ProjectDefinition, list[dict[str, Any]] | None, list[dict[str, Any]]]:
+    project_def = ProjectDefinition(
+        shortcode=project_json["project"]["shortcode"],
+        shortname=project_json["project"]["shortname"],
+        longname=project_json["project"]["longname"],
+        keywords=project_json["project"].get("keywords"),
+        descriptions=project_json["project"].get("descriptions"),
+        groups=project_json["project"].get("groups"),
+        users=project_json["project"].get("users"),
+    )
+
+    # expand the Excel files referenced in the "lists" section of the project (if any), and add them to the project
+    if new_lists := expand_lists_from_excel(project_json.get("project", {}).get("lists", [])):
+        all_lists = new_lists
+    else:
+        all_lists = project_json["project"].get("lists")
+
+    # validate against JSON schema
+    validate_project(project_json, expand_lists=False)
+    print("    JSON project file is syntactically correct and passed validation.")
+    logger.info("JSON project file is syntactically correct and passed validation.")
+
+    all_ontos = project_json["project"]["ontologies"]
+    # rectify the "hlist" of the "gui_attributes" of the properties
+    for onto in all_ontos:
+        if onto.get("properties"):
+            onto["properties"] = _rectify_hlist_of_properties(
+                lists=all_lists or [],
+                properties=onto["properties"],
+            )
+
+    return project_def, all_lists, all_ontos
```

Comment:
I think this function does too many things. An indicator for this is the name: It prepares the project info, the ontos and the lists, so it does 3 things. Another indicator is the long list of returned objects (`tuple[ProjectDefinition, list[dict[str, Any]] | None, list[dict[str, Any]]]`). 

Some ideas how this could be improved:
- put the lists inside the ProjectDefinition
- take the schema validation and the hlist rectification out of this function

Reply:
I have taken the function apart, but regarding the lists in the ProjectDefinition I had the following considerations:
There are three distinct parts of the json: project meta-data, lists and ontologies. In my opinion all of these should be an independent class. If I added the lists to the ProjectDefinition, it would make this class more complex and mix those sections. We already have an ontology class, which in my opinion is quite atrocious. A clean overhaul of this code is definitively a good idea, but I don't think this is that time. Because ideally we could combine it with an implementation of rdflib. So I would also like to wait with a list class until we do a rdflib implementation.
So I decided to do a minimally invasive cleaning.

Reply:
sure, that's fine